doofinder-for-woocommerce/assets/js/wizard.js

```diff
@@ -104,15 +58,65 @@ jQuery(() => {
 
   $(".open-window").click((e) => {
     const pageType = $(e.currentTarget).attr("data-type");
-
     connectModal = popupCenter({
-      url: `https://app.doofinder.com/plugins/${pageType}/woocommerce?email=${doofinderConnectEmail}&token=${doofinderConnectToken}&return_path=${doofinderConnectReturnPath}`,
+      url: `${doofinderAdminPath}/plugins/${pageType}/woocommerce?email=${doofinderConnectEmail}&token=${doofinderConnectToken}`,
       title: "DoofinderConnect",
       w: 600,
       h: 700,
     });
-
-    clearInterval(checkInterval);
-    checkInterval = setInterval(ajaxCheckData, 1000);
   });
+
+  window.addEventListener(
+    "message",
+    (event) => {
+      //Check that the sender is doofinder
+      if (event.origin !== doofinderAdminPath) return;
+      if (event.data) {
+        data = event.data.split("|");
+        event_name = data[0];
+        event_data = JSON.parse(atob(data[1]));
+        processMessage(event_name, event_data);
+      }
+    },
+    false
+  );
+
+  function processMessage(name, data) {
+    switch (name) {
+      case "set_doofinder_data":
+        send_connect_data(data);
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  function send_connect_data(data) {
+    data["action"] = "doofinder_set_connection_data";
+    $.ajax({
+      type: "POST",
+      dataType: "json",
+      url: DoofinderForWC.ajaxUrl,
+      data: data,
+    }).then((response) => {
+      if (response.success) {
+        confirmLeavePage(false);
+        window.location.href = doofinderSetupWizardUrl;
+        return;
+      } else {
+        $(".errors-wrapper ul").remove();
+        if (response.errors) {

```

Comment:
Would it be possible to remove this part to a function so as not to have nested ifs?

Reply:
Yes, good idea, thanks.

Reply:
My suggestion is to refactor this into a promise structure, because it'll be much more clear. The "then" cases, could be managed by the await | async tools, if you want